### PR TITLE
Attempt to add 4th axis

### PR DESCRIPTION
**Pull request by ashelly**
Compiles cleanly for uno 3axis and mega 4axis.  Seems to work in simulator.
Adds 74 bytes to uno 3axis build.   Mega 4axis is 620 bytes bigger than mega 3axis.

Used 'C' for axis name, since it was my "Conveyor".  I suppose 'A' would be a better choice, that should be fairly simple search/replace.

### CODE REVIEW

File: grbl/stepper.c [hunk 1]
```diff
@@ -351,40 +365,44 @@ ISR(TIMER1_COMPA_vect)
...
-    st.counter_z += st.exec_block->steps[Z_AXIS];
-  #endif  
-  if (st.counter_z > st.exec_block->step_event_count) {
+  
+  st.counter[Z_AXIS] += st.steps[Z_AXIS];
+
+  if (st.counter[Z_AXIS] > st.exec_block->step_event_count) {
     st.step_outbits |= (1<<Z_STEP_BIT);
-    st.counter_z -= st.exec_block->step_event_count;
+    st.counter[Z_AXIS] -= st.exec_block->step_event_count;
```
Comment (chamnit) [inline grbl/stepper.c]: @ashelly : Do you know if this compiles out to be an equivalent to st.counter_z? These were explicitly rolled out to get better compiled performance within the ISR. At least, the last time I tested it on the oscilloscope.

**Comment by chamnit** [comment]
@ashelly : I'm going to have to sit on this for a bit. I've got to push fixes for the homing cycle and other things before I get 4th axis installed. I'm also planning on starting a v1.0 dev branch. This would be better served there, since it'll require a lot of testing, vetting, and bug fixes to ensure a stable general release. But, I'll take what you have here and use it as a framework.

File: grbl/stepper.c (hunk 1 above)
Comment (ashelly) [inline grbl/stepper.c]: I did not check the assembly.  Making an array allowed cleaner looking c
code than the alternative with multiple #ifdefs.     But I agree
performance overrules style considerations. I will take a look later.

**Comment by ashelly** [comment]
I don't mind at all.  I just wanted to put my solution out there,  and save
you the trouble of picking it out of my bigger set of changes.  I know it
needs significant testing.

If you have higher priority features or issues you can use help with in the
short term,  let me know.

**Comment by mkeyno** [comment]
hi guys, I have same question , actually I decide to engrave image or letter on the metal ring , I've just intend to aligned my penomatic hammer with central line of rotary axis and use only axis A & Y , any help or links really appreciated

**Pull request closed without merging**